data_transfer/main.py

```diff
@@ -22,4 +30,9 @@
     if device == DeviceType.SMA:
         sma.dag()
     if device == DeviceType.BTF:
-        btf.dag(study_site)
+        # if 'days' == -1, trigger full history
+        if len(sys.argv) >= 4 and int(sys.argv[3]) == -1:
+            btf.historical_dag(study_site, *sys.argv[3:])
+        else:
+            # passes timespan and reference if present
```

Comment:
Before `if len(...)`  I would define the variables as arguments and cast them once as needed. This mirrors the potential use of `click` or other CLIs where arguments are explictely defined. This also means you can cast those once and define intended input in both `dag` and `historical_dag` rather than using `Union` or such

Reply:
Am fixing this as we speak - but note that using `None` does not allow picking up the default and will require additional logic in the methods itself. 

Also, leaving out `click` implementation due to priorities